util/testkit/testkit.go

```diff
@@ -251,6 +252,16 @@ func (tk *TestKit) ExecToErr(sql string, args ...interface{}) error {
 	return err
 }
 
+// MustGetErrCode executes a sql statement and assert it's error code of the error.
+func (tk *TestKit) MustGetErrCode(sql string, errCode int) {
+	_, err := tk.Exec(sql)
+	tk.c.Assert(err, check.NotNil)
+	originErr := errors.Cause(err)
+	tErr, ok := originErr.(*terror.Error)
+	tk.c.Assert(ok, check.IsTrue, check.Commentf("err: %T", originErr))
+	tk.c.Assert(tErr.ToSQLError().Code, check.Equals, uint16(errCode), check.Commentf("MySQL code:%v", tErr.ToSQLError()))
+}
+
```

Comment:
Could we reuse this function? e.g. extra this function to a shared package
https://github.com/pingcap/tidb/blob/d9e7bd315ed3494ece6879186503e891934973b8/ddl/db_test.go#L140-L147

Reply:
@Deardrops I think `MustGetErrCode` is easier to use than `assertErrorCode`. Later I will remove `assertErrorCode` in another PR.